rholang/src/main/scala/coop/rchain/rholang/interpreter/RholangCLI.scala

```diff
@@ -24,34 +25,51 @@ object RholangCLI {
     verify()
   }
 
-  def reader(fileName: String): FileReader = new FileReader(fileName)
-  def lexer(fileReader: Reader): Yylex     = new Yylex(fileReader)
-  def parser(lexer: Yylex): parser         = new parser(lexer, lexer.getSymbolFactory())
-
   def main(args: Array[String]): Unit = {
     val conf = new Conf(args)
     if (conf.file.supplied) {
       val fileName: String        = conf.file()
       val source                  = reader(fileName)
       val sortedTerm: Option[Par] = buildNormalizedTerm(source)
+
       if (conf.binary()) {
         writeBinary(fileName, sortedTerm.get)
       } else {
         writeHumanReadable(fileName, sortedTerm.get)
       }
+      // TODO: In case we want to print out result from the evaluator for the CLI
+      // uncomment and adjust the following.
+      //
+      // Evaluator.run(sortedTerm.get)
     } else {
       print("> ")
       repl
     }
   }
 
+  private def reader(fileName: String): FileReader = new FileReader(fileName)
+  private def lexer(fileReader: Reader): Yylex     = new Yylex(fileReader)
+  private def parser(lexer: Yylex): parser         = new parser(lexer, lexer.getSymbolFactory())
+
+  // TODO: And delete this
+  private def printTask(normalizedTerm: Par): Task[Unit] =
+    Task now {
+      PrettyPrinter.prettyPrint(normalizedTerm)
+      print("\n> ")
+    }
+
   private def repl =
     for (ln <- Source.stdin.getLines) {
-      print(buildNormalizedTerm(new StringReader(ln)).get.toString)
-      print("> ")
+      if (ln.isEmpty) {
+        print("> ")
+      } else {
+        val normalizedTerm = buildNormalizedTerm(new StringReader(ln)).get
+        // TODO: Replace below with Evaluator.run(sortedTerm.get)
+        printTask(normalizedTerm)
```

Comment:
Doesn't this mean that repl is now of type Task[Unit]?
You don't have anything driving the Task from the main loop. I'm surprised this even works.

Reply:
Indeed. It works because Task now executes it immediately.

Reply:
Resolved through zoom call.